src/flamegraph/color/palettes.rs

```diff
@@ -59,6 +59,21 @@ pub(super) mod perl {
     }
 }
 
+pub(super) mod python {
+    use crate::flamegraph::color::BasicPalette;
+
+    pub fn resolve(name: &str) -> BasicPalette {
+        if name.starts_with("native@") { // austin-specific format for native calls
```

Comment:
I worry about encoding generator-specific detection into here. Is there no other way to detect native calls that isn't austin-specific?

Reply:
I agree it's a bit nasty to be doing profiler specific checks here. Unfortunately since getting profiles which combine Python stack traces with native calls is [pretty involved](https://www.benfrederickson.com/profiling-native-python-extensions-with-py-spy/), we are at the mercy of how the profiler decides to format things.

As far as I know the only profilers which support native calls are austin and py-spy, so if we can support both of those then maybe it's fine.

All austin native calls are prefixed with `native@`, e.g.
```
native@7f8b5cbb69a6:ufunc_generic_fastcall:12022
```

py-spy looks like this:
```
ufunc_generic_fastcall (numpy/core/_multiarray_umath.cpython-39-x86_64-linux-gnu.so)
```

I've tagged the author of py-spy in this PR, it would be good to get his input on the best way of parsing this. We could easily do a check for `.so` or `.dll`, but I suspect that if the native extension is built with debug symbols then we would get a reference to a source file, which could have any extension.

Alternatively, I'm happy to remove this branch and ignore native calls for now. For me at least it is very rare to need to go down into native calls when profiling Python.

Reply:
I'm really torn here. I'd love to hear from the py-spy maintainer too, but I think given the state of the landscape I'm then okay with putting in place checks specifically for austin/py-spy's markers for native calls.